server/store/subscriptions.go

```diff
@@ -106,6 +106,7 @@ func (s *Store) GetAllSubscriptions(userID string) ([]*serializers.SubscriptionD
 
 	var subscriptionList []*serializers.SubscriptionDetails
 	for _, subscription := range subscriptions.ByMattermostUserID[userID] {
+		subscription := subscription
```

Comment:
You should also add a comment above this line specifying why you are doing this.